src/controllers/socialController.js

```diff
@@ -0,0 +1,102 @@
+import jwt from 'jsonwebtoken';
+import {
+  config
+} from 'dotenv';
+import db from '../models';
+
+config();
+
+class SocialController {
+  static async findOrCreateUserByEmail(firstName, lastName, email) {
+    const [{
+      dataValues: userData
+    }] = await db.User.findOrCreate({
+      defaults: {
+        firstName,
+        lastName,
+        email
+      },
+      where: {
+        email
+      }
+    });
+    return userData;
+  }
```

Comment:
Good job here. I thought your feature was to log in a user with google or facebook not to create a user. That will involve a lot more logic and code. so I suggest you stick to login in a user with google or facebook. Also, remember to verify that a user's email has been verified before logging the user in.

Reply:
so what you are doing is to check if the email from google or facebook has a corresponding email in the db, then while querying you return the `isVerified` column. to check if the user has been verified.